knotx-core/src/main/java/io/knotx/proxy/reactive/AdapterProxyFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.proxy.reactive;
+
+
+import io.knotx.reactivex.proxy.AdapterProxy;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.reactivex.core.Vertx;
+import java.util.Optional;
+
+public final class AdapterProxyFactory {
+
+  private AdapterProxyFactory() {
+    // hidden
+  }
+
+  public static AdapterProxy createProxy(Vertx vertx, Optional<DeliveryOptions> deliveryOptions, String address) {
```

Comment:
Not sure how this fits into the architecture of this area of Knot.x but consuming an Optional as a parameter seems a bit strange. How about overloading the method to either accept the `DeliveryOptions` or not?

    public static AdapterProxy createProxy(Vertx vertx, String address, DeliveryOptions deliveryOptions)
    //...
    public static AdapterProxy createProxy(Vertx vertx, String address)

The caller either has the delivery options or not. If they don't, with the current API, they need to pass an empty optional here.

I can see this pattern in more places and I'm wondering what the benefits of using an `Optional` this way are.

Reply:
@toniedzwiedz Please check what is the expected behavior in this case. We have a configuration that has a getter:
```Optional<DeliveryOptions> get()```
It means that we expect DeliveryOptions may be not defined. Based on this value we want to create a factory method that uses two different methods with API (with and without DeliveryOptions). I don't know what is best in this situation...